x/storage/keeper/keeper.go

```diff
@@ -2230,13 +2230,23 @@ func (k Keeper) SetTag(ctx sdk.Context, operator sdk.AccAddress, grn types2.GRN,
 		if !found {
 			return types.ErrNoSuchBucket.Wrapf("bucketName: %s", bucketName)
 		}
-		resOwner := sdk.MustAccAddressFromHex(bucketInfo.Owner)
-		if !operator.Equals(resOwner) {
-			return types.ErrAccessDenied.Wrapf(
-				"Only resource owner can set tag, operator (%s), owner(%s)",
-				operator.String(), resOwner.String())
-		}
 
+		// todo define a new hardfork
+		if ctx.IsUpgraded(upgradetypes.Hulunbeier) {
+			// check permission
+			effect := k.VerifyBucketPermission(ctx, bucketInfo, operator, permtypes.ACTION_UPDATE_BUCKET, nil)
```

Comment:
Does the owner have the permission in this case?

And I think we dont have to add a new hardfork?  As this patch is with v1.4.0, so reuse the hardfork of v1.4.0.

Reply:
yes, the owner has all permissions. And v1.4.0 hardfork is not defined yet?